docs/proposals/error-handling-guideline.md

```diff
@@ -0,0 +1,225 @@
+
+## How to write friendly error message
+
+### Recommended error meesage structure
+
+Here is a sample structure of an error message:
+
+```text
+Error: [Error description] : [Error code] ：[Error title]
+
+Usage: [Sample command]
+Help: [Recommended solution], [TSG]
+```
+
+### Examples
+
+Here are some examples of writing error message with helpful prompt actionable information:
+
+#### Example 1: when no reference provided in `oras copy`
+
+Current behavior and output:
+
+```
+$ oras cp
+Error: accepts 2 arg(s), received 0
+```
+
+Suggested error message:
+
+```
+$ oras cp
+Error: "oras copy" requires exactly 2 arguments.
+
+Usage: oras copy [flags] <from>{:<tag>|@<digest>} <to>[:<tag>[,<tag>][...]]
+Help: Copy artifacts from one target to another. Run "oras copy -h" for more options and examples
+```
+
+#### Example 2: when reference does not match with the expected format
+
+Current behavior and output:
+
+```
+$ oras tag list ghcr.io/oras-project/oras
+Error: unable to add tag for 'list': invalid reference: missing repository
+```
+
+Suggested error message:
+
+```
+$ oras tag list ghcr.io/oras-project/oras
+Error: unable to add tag for 'list': invalid reference: missing repository
+
+Usage: oras tag [flags] <name>{:<tag>|@<digest>} <new_tag> [...]
+Help: Tag a manifest in a registry or an OCI image layout. Run "oras tag -h" for more options and examples
+```
+
+#### Example 3: When fetching a manifest if no manifest tag or digest is provided
+
+Current behavior and output:
+
+```
+$ oras manifest fetch --oci-layout /tmp/ginkgo1163328512 >>
+Error: "/tmp/ginkgo1163328512": no tag or digest when expecting <name:tag|name@digest>
+```
+
+Suggested error message:
+
+```
+$ oras manifest fetch --oci-layout /tmp/ginkgo1163328512 >>
+Error: "/tmp/ginkgo1163328512": no tag or digest specified
+
+Usage: oras manifest fetch [flags] <name>{:<tag>|@<digest>}
+Help: Fetch manifest of the target artifact. Run "oras manifest fetch -h" for more options and examples 
+```
+
+#### Example 4: push a manifest if no media type flag provided
+
+Current behavior and output:
+
+```
+$ oras manifest push --oci-layout /tmp/ginkgo2167255592:mediatype-flag
+Error: media type is not recognized. 
+```
+
+Suggested error message:
+
+```
+$ oras manifest push --oci-layout /tmp/ginkgo2167255592:mediatype-flag
+Error: media type is not recognized. Specify an valid media type with "--media-type"
+```
+
+#### Example 5: attach an artifact if the given option is unknown
+
+Current behavior and output:
+
+```
+$ oras attach --artifact-type oras/test localhost:7000/command/images:foobar --distribution-spec ???
+Error: unknown distribution specification flag: "???"
+```
+
+Suggested error message:
+
+```
+$ oras attach --artifact-type oras/test localhost:7000/command/images:foobar --distribution-spec ???
+Error: unknown distribution specification flag: "???". Available options: v1.1-referrers-api, v1.1-referrers-tag
```

Comment:
`Available options: v1.1-referrers-api, v1.1-referrers-tag` Should we put this to `Help:`?